### ISSUE TITLE
Unable to get StringPatriciaMap to work with utf-8 keys

### ISSUE THREAD

**Issue by jannisbecker** [sile/patricia_tree#35]
Hello!

I'm trying to build a japanese dictionary app with the StringPatriciaMap as a searchable index for searching the dictionary quickly.

Given the following mini example, this does not work:
```rust
    let mut map = StringPatriciaMap::<u8>::new();
    map.insert("インターポール", 1);
    map.insert("インターポル", 2);
    map.insert("インターリーブ", 3);
    map.insert("インターン", 4);

    assert_eq!(map.get("インターポール"), Some(&1)); //works
    assert_eq!(map.get("インターポル"), Some(&2)); //doesn't work
```

map.get will return Some if I search for the **first** key, but None if I search for the second or any other key.

This might just be a misunderstanding on how this data structure works, so please enlighten me if that's the case.

**Comment by sile** [comment]
Thank you for reporting this issue.
I'm not certain, but it appears to be a bug within this crate.
I will conduct a more thorough investigation when I have the time, likely within a week.

**Comment by sile** [comment]
Fixed this issue and just released v0.7.0.
Thanks again for reporting this bug!
